src/components/Table.jsx

```diff
@@ -143,31 +115,119 @@ export default function Table() {
           assignee: !isAdmin ? userEmail : searchByAssignee
         }),
         ...(searchByUpdatedAt && { lastUpdateTs: finalDate }),
-        status: isStatusSelected ? [searchByStatus] : status
+        status: isStatusSelected && !clearAllFilters ? [searchByStatus] : status
       },
       orderBy: {
-        ...(currentSort == "styleSort" && { styleId: styleSort }),
-        ...(currentSort == "titleSort" && { title: titleSort }),
-        ...(currentSort == "brandSort" && { brand: brandSort }),
-        ...(currentSort == "updatedBySort" && { lastUpdatedBy: updatedBySort }),
-        ...(currentSort == "assigneeSort" && { lastUpdatedBy: assigneeSort }),
-        ...(currentSort == "statusSort" && { status: statusSort }),
-        ...(currentSort == "updatedAtSort" && { lastUpdateTs: updatedAtSort })
+        ...(currentSort == "Style" && { styleId: styleSort }),
+        ...(currentSort == "Title" && { title: titleSort }),
+        ...(currentSort == "Brand" && { brand: brandSort }),
+        ...(currentSort == "Updated By" && {lastUpdatedBy : updatedBySort }),
+        ...(currentSort == "Assignee" && { assignee : assigneeSort }),
+        ...(currentSort == "Status" && { status: statusSort }),
+        ...((currentSort == "Updated At" || (currentSort != "Style" && currentSort != "Title" && currentSort != "Brand" && currentSort != "Updated By" && currentSort != "Assignee" && currentSort != "Status" )) && {lastUpdateTs :  updatedAtSort })
       }
     };
-    setLoader(true);
-    fetch(`${workFlowsUrl}/search?limit=10&page=${currentPage}`, {
-      method: "POST",
-      body: JSON.stringify(body),
-      headers: {
-        "Content-type": "application/json; charset=UTF-8"
+      const response = await fetch(`${workFlowsUrl}/search?limit=10&page=${currentPage}`,
+       {
+        method: 'POST',
+        body: JSON.stringify(body),
+        headers: {
+          "Content-type": "application/json; charset=UTF-8"
+        }
       }
-    })
-      .then((response) => response.json())
-      .then((result) => {
-        setCustomers(result?.data);
-        setLoader(false);
-      });
+  );
+  const data = await response.json();
+  if(data?.success){
+    setCustomers(data?.data);
+    setLoader(false)
+  }
+  if (!data?.success){
+    console.log("data>>",data?.error)
+	}
+}
+
+async function getBrands(){
+const response = await fetch(`${workFlowsUrl}/search?limit=${limit}&page=${currentPage}&unique=brand`,
```

Comment:
why is this approach not implemented in other places.
I can still see some fetch calls with then

Reply:
addressed